File: cmsong111/42842.py
def solution(brown, yellow):
    max = brown//2 + 2

    for i in range(max-3,2,-1):
        if (i-2) * (max-i-2) == yellow:
            return [i,max-i]  


brown = 10
yellow = 2	
print(solution(brown, yellow)) 
# answer = [4, 3]

brown = 8	
yellow = 1
print(solution(brown, yellow)) 
# answer = [3, 3]

brown = 24	
yellow = 24	
print(solution(brown, yellow))
# answer = [3, 3]

File: sudhdkso/42842.cpp
#include <string>
#include <vector>

using namespace std;

vector<int> solution(int brown, int yellow) {
    vector<int> answer;
    int area = brown + yellow;
    int h=0;
    for (int w = area - 1; w > 0; w--) {
        if (area % w) continue;

        h = area / w;
        int y_area = (h - 2) * (w - 2);
        int b_area = area - y_area;

        if (y_area == yellow && b_area == brown) {
            answer.push_back(w);
            answer.push_back(h);
            return answer;
        }

    }
    
    return answer;
}